--- a/ext/standard/php_fopen_wrapper.c
+++ b/ext/standard/php_fopen_wrapper.c
@@ -318,7 +318,11 @@ php_stream * php_stream_fopen_from_fd_int_rel(int fd, const char *mode, const c
 		}
 
 #ifdef HAVE_UNISTD_H
+#ifdef __APPLE__  /* iOS uses sysconf(_SC_OPEN_MAX) instead of getdtablesize() */
+		dtablesize = sysconf(_SC_OPEN_MAX);
+#else
 		dtablesize = getdtablesize();
+#endif
 #else
 		dtablesize = INT_MAX;
 #endif
